packages/components/src/tools-panel/tools-panel/hook.ts

```diff
@@ -167,24 +174,24 @@ export function useToolsPanel(
 	] = useState( false );
 
 	useEffect( () => {
-		if ( menuItems.optional ) {
-			const optionalItems = Object.entries( menuItems.optional );
-			const allControlsHidden =
-				optionalItems.length > 0 &&
-				! optionalItems.some( ( [ , isSelected ] ) => isSelected );
+		if (
+			isMenuItemTypeEmpty( menuItems?.default ) &&
```

Comment:
Now we've added `menuIitems.default` as a dependency for this `useEffect` we should probably add it to the hooks dependency array, change the dependency to only `menuItems`, or determine this separately and pass only the boolean result to the hook.